src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -378,9 +431,23 @@ private String getHeaderValueImpl (String uri, String method) {
                 ncstring = zeroPad [len] + ncstring;
         }
 
+        boolean session = algorithm.endsWith ("-sess");
+        if (session) {
+            algorithm = algorithm.substring(0, algorithm.length() - 5);
+        }
+
+        try {
+            validateAlgorithm(algorithm);
+        } catch (IOException e) {
+            // should not happen since the algorithm has already been
+            // validated
```

Comment:
Hello Michael, was this comment meant for something else? The comments feels a bit odd since it says "has already been validated" which isn't the case since the `validateAlgorithm` itself has failed here.

Reply:
Yes, copy and paste error. Thanks